src/theory/quantifiers/sygus/cegis_unif.cpp

```diff
@@ -135,6 +138,22 @@ bool CegisUnif::processConstructCandidates(const std::vector<Node>& enums,
         // get the current unification enumerators
         d_u_enum_manager.getEnumeratorsForStrategyPt(
             e, unif_enums[index][e], index);
+        if (index == 1 && options::sygusUnifCondIndependent())
+        {
+          Assert(unif_enums[index][e].size() == 1);
+          Node eu = unif_enums[index][e][0];
+          Assert(d_u_enum_manager.d_enum_to_active_guard.find(eu)
+                 != d_u_enum_manager.d_enum_to_active_guard.end());
+          Node g = d_u_enum_manager.d_enum_to_active_guard[eu];
```

Comment:
Can you put this logic in the EnumManager?  Just take this code and make it a function in that class and call it here.  This will be needed for the decision manager refactoring.